.github/workflows/cd.yml

```diff
@@ -8,7 +8,11 @@ jobs:
     name: CD
     runs-on: ubuntu-latest
     environment: CD
+    permissions:
+      contents: read
+      id-token: write
```

Comment:
Why is this block needed?

Reply:
The content read, well it doesn't do much, just allow a read access to the commits etc.
The id-token is a requirement for the provenance signature.